### ISSUE TITLE
compile error

### ISSUE THREAD

**Issue by zychaoqun** [dfridovi/exploration#7]
Dear Sir,
I have try to install your package following your instruction. The first error:

`/home/chaoqun/exploration/radiation/cpp/exec/first_gl_program/main.cpp:46:56: fatal error: GLUT/glut.h: No such file or directory
 #include <GLUT/glut.h>  // GLUT, include glu.h and gl.h`

I tried to change the include format into "#include<GL/glut.h>", this problem seems to be solved.

And I encountered another problem:

`/home/chaoqun/exploration/radiation/cpp/exec/run_explorer_lp/main.cpp:146:3: error: ‘gflags’ has not been declared
   gflags::ParseCommandLineFlags(&argc, &argv, true);
   ^
`
I change 'gflags' to 'google', this problem also seems to be solved.

Now another problem appeared:
`/home/chaoqun/exploration/radiation/cpp/exec/run_explorer_lp/main.cpp:146:3: error: ‘gflags’ has not been declared
   gflags::ParseCommandLineFlags(&argc, &argv, true);
`
Now I have no idea what I should do ,Any suggestions?

**Comment by dfridovi** [comment]
Wow -- that's a lot of compiler issues. Can I ask what your system is (i.e. OS, compiler, etc.)? I develop on a Mac running OS X El Capitan and the clang compiler, so I'm guessing something is different on your system. 

I have run into the GLUT/GL header issue before, and also the `gflags` vs `google` namespace issue. My suggestion would be to again replace `gflags::ParseCommandLineFlags(...)` with `google::ParseCommandLineFlags(...)`. If that doesn't work, maybe you can give a more comprehensive description of the error (i.e. the full compiler output)?

**Comment by zychaoqun** [comment]
I am using cmake on Linux 14.04,can I achieve using this version Ubuntu？
I have solved the problem caused by the head file, now there is only one problem left:

> /home/chaoqun/exploration/radiation/cpp/exec/run_explorer_lp/main.cpp:146:3: error: ‘gflags’ has not been declared gflags::ParseCommandLineFlags(&argc, &argv, true);

Little confused how to solved this.

**Comment by zychaoqun** [comment]
I am using cmake on Linux 14.04,can I achieve using this version Ubuntu？
I have solved the problem caused by the head file, now there is only one problem left:

`/usr/bin/ld: /usr/local/lib/libceres.a(evaluator.cc.o): undefined reference to symbol 'pthread_rwlock_wrlock@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
`
Little confused how to solved this.

Thanks for your help!

**Comment by dfridovi** [comment]
Hmm ok this one looks like an issue with your build of `Ceres`. My guess is that you installed `Ceres` from `apt`. I would suggest uninstalling it on `apt` and reinstalling it from source (just clone the ceres repo and build it like any other cmake project).

**Comment by zychaoqun** [comment]
Thank you for your reply!
I have add the `${CMAKE_THREAD_LIBS_INIT}` in  /home/chaoqun/exploration/radiation/cpp/CMakeList
`target_link_libraries(${executable} radiation ${radiation_LIBRARIES} ${CERES_LIBRARY}  ${CMAKE_THREAD_LIBS_INIT} ${CHOLMOD_LIBRARIES}`)
Now the above problems seems to be solved. But ,new problem appeared:
reorder_program.cc:(.text+0x1b7): undefined reference to `cholmod_free_sparse'
`/usr/local/lib/libceres.a(reorder_program.cc.o): In function `ceres::internal::(anonymous namespace)::OrderingForSparseNormalCholeskyUsingSuiteSparse(ceres::internal::TripletSparseMatrix const&, std::vector<ceres::internal::ParameterBlock*, std::allocator<ceres::internal::ParameterBlock*> > const&, ceres::OrderedGroups<double*> const&, int*)':`

I have installed ceres from source but the problem still exists. Any idea? Thanks!

**Comment by dfridovi** [comment]
Yeah so these are all errors related to sparse solving methods in ceres. I'm not too familiar with them, since usually the problems I use it for are dense, but in any case you shouldn't need them here. You can configure the ceres build (by using the `ccmake` gui, e.g.) to not build sparse solver stuff and maybe this will go away...

Another thing related to the `pthread` compiler error above -- maybe try installing `pthread` as follows
```
sudo apt-get install libpthread-stubs0-dev
```
before re-compiling ceres.
